server/src/test/java/org/elasticsearch/action/admin/indices/create/ShrinkIndexIT.java

```diff
@@ -554,4 +556,46 @@ public void testShrinkCommitsMergeOnIdle() throws Exception {
             EnableAllocationDecider.CLUSTER_ROUTING_REBALANCE_ENABLE_SETTING.getKey(), (String)null
         )).get();
     }
+
+    public void testShrinkThenSplitWithFailedNode() throws Exception {
+        internalCluster().ensureAtLeastNumDataNodes(3);
+
+        final int shardCount = between(2, 5);
+        prepareCreate("original").setSettings(Settings.builder().put(indexSettings())
+            .put(IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, shardCount)).get();
+        for (int i = 0; i < 30; i++) {
+            client().prepareIndex("original", "_doc")
+                .setSource("{\"foo\" : \"bar\", \"i\" : " + i + "}", XContentType.JSON).get();
```

Comment:
NIT: `.setSource("foo", "bar", "i" , i)` seems nicer? :)

Reply:
b1803fc459c is nicer still ;)